airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -114,78 +120,110 @@ private void runInternal(final IntegrationConfig parsed) throws Exception {
-            produceMessages(messageIterator);
-          }
-        }
+        case READ -> runRead(parsed);
         // destination only
-        case WRITE -> {
-          final JsonNode config = parseConfig(parsed.getConfigPath());
-          validateConfig(integration.spec().getConnectionSpecification(), config, "WRITE");
-          final ConfiguredAirbyteCatalog catalog = parseConfig(parsed.getCatalogPath(), ConfiguredAirbyteCatalog.class);
-          try (final AirbyteMessageConsumer consumer = destination.getConsumer(config, catalog, outputRecordCollector)) {
-            runConsumer(consumer);
-          }
-        }
+        case WRITE -> runWrite(parsed);
         default -> throw new IllegalStateException("Unexpected value: " + parsed.getCommand());
       }
     } catch (final Exception e) {
-      // Many of the exceptions thrown are nested inside layers of RuntimeExceptions. An attempt is made
-      // to
-      // find the root exception that corresponds to a configuration error. If that does not exist, we
-      // just return the original exception.
-      ApmTraceUtils.addExceptionToTrace(e);
-      final Throwable rootThrowable = ConnectorExceptionUtil.getRootConfigError(e);
-      final String displayMessage = ConnectorExceptionUtil.getDisplayMessage(rootThrowable);
-      // If the source connector throws a config error, a trace message with the relevant message should
-      // be surfaced.
-      if (ConnectorExceptionUtil.isConfigError(rootThrowable)) {
-        AirbyteTraceMessageUtility.emitConfigErrorTrace(e, displayMessage);
-      }
-      if (parsed.getCommand().equals(Command.CHECK)) {
-        // Currently, special handling is required for the CHECK case since the user display information in
-        // the trace message is
-        // not properly surfaced to the FE. In the future, we can remove this and just throw an exception.
-        outputRecordCollector
-            .accept(
-                new AirbyteMessage()
-                    .withType(Type.CONNECTION_STATUS)
-                    .withConnectionStatus(
-                        new AirbyteConnectionStatus()
-                            .withStatus(AirbyteConnectionStatus.Status.FAILED)
-                            .withMessage(displayMessage)));
-        return;
-      }
-      throw e;
+      handleRunInternalException(parsed, e);
     }
 
     LOGGER.info("Completed integration: {}", integration.getClass().getName());
   }
 
+  private AirbyteMessage runCheck(final IntegrationConfig parsed) throws Exception {
+    try {
+      final JsonNode config = getValidatedConfig(parsed, CHECK.toString());
+      return new AirbyteMessage().withType(Type.CONNECTION_STATUS).withConnectionStatus(integration.check(config));
+    } catch (final ConfigValidationException e) {
+      // if validation fails don't throw an exception, return a failed connection check message
+      return failedConnectionStatusMessage(e.getMessage());
```

Comment:
we're discarding the full stacktrace here - let'es at least log `e`. Maybe also worth emitting an AirbyteTraceMessage message with the stacktrace too

Reply:
or is the idea that the read/write/etc methods are supposed to do that already?

Reply:
I think Logging the exception is a good idea. I was trying to preserve the existing behavior as much as possible, in the previous implementation the check method did this after a call to `validateConfig` which might throw an exception (which also swallows the full stacktrace)
```
 catch (final Exception e) {
            // if validation fails don't throw an exception, return a failed connection check message
            outputRecordCollector.accept(new AirbyteMessage().withType(Type.CONNECTION_STATUS).withConnectionStatus(
                new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.FAILED).withMessage(e.getMessage())));
          }
```

So if there are no concerns with adjusting this behavior, I'm in favor of at least logging it.